File: frontend/src/types/notifications.ts
export interface NotificationLogEntryResponse {
  id: number
  notification_key: string
  channel: string
  subject: string
  body: string
  sent_at: string
  status: string
}

export interface NotificationCheckResponse {
  new_entry_count: number
  entries: NotificationLogEntryResponse[]
}

File: src/api/schemas/notifications.py
"""API schemas for notification endpoints."""

from pydantic import BaseModel


class NotificationLogEntryResponse(BaseModel):
    """Serialized representation of one notification log entry."""

    id: int
    notification_key: str
    channel: str
    subject: str
    body: str
    sent_at: str
    status: str


class NotificationCheckResponse(BaseModel):
    """Result of an on-demand notification check."""

    new_entry_count: int
    entries: list[NotificationLogEntryResponse]
